app/logic/Calendar/Event.ts

```diff
@@ -772,33 +773,36 @@ export class Event extends Observable {
   }
 
   /**
-   * Deletes the event and any subsequent instances that are not exceptions.
+   * Deletes this instance or exception, and all after that.
    */
-  async truncateRecurrence() {
+  truncateRecurrencesFromHere() {
+    assert((this.recurrenceCase == RecurrenceCase.Instance || this.recurrenceCase == RecurrenceCase.Exception) &&
+      this.parentEvent?.recurrenceRule, "Must be a recurring instance");
     let master = this.parentEvent;
-    assert(master.recurrenceCase == RecurrenceCase.Master, "parentEvent must a Recurrence.Master");
-    let lastException = this.recurrenceStartTime;
-    for (let exception of master.exceptions) {
-      if (exception.recurrenceStartTime > lastException) {
-        lastException = exception.recurrenceStartTime;
-      }
-    }
-    master.generateRecurringInstances(lastException);
-    let exclusions = master.instances.contents.filter(event => event.recurrenceStartTime> this.recurrenceStartTime && event.recurrenceStartTime < lastException);
-    if (lastException > this.recurrenceStartTime) {
-      // Isn't cache invalidation fun!
-      exclusions.push(master.getOccurrenceByDate(this.recurrenceStartTime));
-    }
-    let count = master.recurrenceRule.getIndexOfOccurrence(lastException) + (lastException > this.recurrenceStartTime ? 1 : 0);
-    if (master.recurrenceRule.getOccurrenceByIndex(count)) {
-      await master.setRecurrenceCount(count);
-    }
-    if (exclusions.length) {
-      await master.makeExclusions(exclusions);
-    } else if (!this.isNew) {
-      await this.calendar.storage.deleteEvent(this);
-    }
-    await master.save();
+    let index = master.recurrenceRule.getIndexOfOccurrence(this.recurrenceStartTime);
+    master.setRecurrenceCount(index);
+  }
+
+  /** Creates a new series with the same properties and recurrence,
+   * but starting at this instance.
+   *
+   * It will also truncate the old series at that start time.
+   *
+   * @param start From when on the new series should start
+   * @returns the new master event
+   */
+  cloneSeriesFromThisInstance(): Event {
+    assert((this.recurrenceCase == RecurrenceCase.Instance || this.recurrenceCase == RecurrenceCase.Exception) &&
+      this.parentEvent?.recurrenceRule, "Must be a recurring instance");
+    let oldMaster = this.parentEvent;
+    let newMaster = this.calendar.newEvent();
+    newMaster.startEditing();
+    newMaster.copyEditableFieldsFrom(this);
+    newMaster.calUID = null;
+    let { frequency, interval, week, weekdays } = oldMaster.recurrenceRule;
+    newMaster.newRecurrenceRule(frequency, interval, week, weekdays);
```

Comment:
`oldMaster.recurrenceRule` could be null if you changed the frequency in the UI first. (This would replace the remainder of the series with a single event.)